### PR TITLE
obs-ffmpeg,cmake: Replace external folder by obs-deps headers

### PR DESCRIPTION
**Pull request by tytan652**
### Description

Remove external folder from `obs-ffmpeg` and use obs-deps headers.

### Motivation and Context

Cleanup

### How Has This Been Tested?

CI build.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: cmake/Modules/FindFFnvcodec.cmake [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#[=======================================================================[.rst
+FindFFnvcodec
+----------
```
Comment (RytoEX) [inline cmake/Modules/FindFFnvcodec.cmake:3]: Make this the same length as the text above.

File: cmake/finders/FindFFnvcodec.cmake [hunk 2]
```diff
@@ -0,0 +1,96 @@
+#[=======================================================================[.rst
+FindFFnvcodec
+----------
```
Comment (RytoEX) [inline cmake/finders/FindFFnvcodec.cmake:3]: Make this underline the same length as the text above.

File: cmake/finders/FindFFnvcodec.cmake [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  if(NOT TARGET FFnvcodec::FFnvcodec)
+    add_library(FFnvcodec::FFnvcodec INTERFACE IMPORTED)
+    set_target_properties(FFnvcodec::FFnvcodec PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${FFnvcodec_INCLUDE_DIR}")
+  endif()
+endif()
+
+include(FeatureSummary)
+set_package_properties(
+  FFnvcodec PROPERTIES
+  URL "https://git.videolan.org/?p=ffmpeg/nv-codec-headers.git"
```
Comment (RytoEX) [inline cmake/finders/FindFFnvcodec.cmake:95]: You may consider using their GitHub mirror. Though if this is just informative and not being used to fetch contents, then it doesn't matter.

File: cmake/Modules/FindFFnvcodec.cmake [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  if(NOT TARGET FFnvcodec::FFnvcodec)
+    add_library(FFnvcodec::FFnvcodec INTERFACE IMPORTED)
+    set_target_properties(FFnvcodec::FFnvcodec PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${FFnvcodec_INCLUDE_DIR}")
+  endif()
+endif()
+
+include(FeatureSummary)
+set_package_properties(
+  FFnvcodec PROPERTIES
+  URL "https://git.videolan.org/?p=ffmpeg/nv-codec-headers.git"
```
Comment (RytoEX) [inline cmake/Modules/FindFFnvcodec.cmake:95]: You may consider using their GitHub mirror. Though if this is just informative and not being used to fetch contents, then it doesn't matter.

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: cmake/Modules/FindFFnvcodec.cmake (hunk 1 above)
Comment (tytan652) [reply cmake/Modules/FindFFnvcodec.cmake]: Done.

File: cmake/finders/FindFFnvcodec.cmake (hunk 3 above)
Comment (tytan652) [reply cmake/finders/FindFFnvcodec.cmake]: Done.

File: cmake/finders/FindFFnvcodec.cmake (hunk 2 above)
Comment (tytan652) [reply cmake/finders/FindFFnvcodec.cmake]: Done.

File: cmake/Modules/FindFFnvcodec.cmake (hunk 4 above)
Comment (tytan652) [reply cmake/Modules/FindFFnvcodec.cmake]: Done.

**Comment by tytan652** [comment]
Licensed finder under MIT.

**Review by PatTheMav — APPROVED** [review]

File: cmake/Modules/FindFFnvcodec.cmake [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    add_library(FFnvcodec::FFnvcodec INTERFACE IMPORTED)
+    set_target_properties(FFnvcodec::FFnvcodec PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${FFnvcodec_INCLUDE_DIR}")
+  endif()
+endif()
+
+include(FeatureSummary)
+set_package_properties(
+  FFnvcodec PROPERTIES
+  URL "https://github.com/FFmpeg/nv-codec-headers/"
+  DESCRIPTION "FFmpeg version of headers required to interface with Nvidias codec APIs.")
```
Comment (RytoEX) [inline cmake/Modules/FindFFnvcodec.cmake:96]:
```suggestion
  DESCRIPTION "FFmpeg version of headers required to interface with NVIDIA's codec APIs.")
```

File: cmake/finders/FindFFnvcodec.cmake [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    add_library(FFnvcodec::FFnvcodec INTERFACE IMPORTED)
+    set_target_properties(FFnvcodec::FFnvcodec PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${FFnvcodec_INCLUDE_DIR}")
+  endif()
+endif()
+
+include(FeatureSummary)
+set_package_properties(
+  FFnvcodec PROPERTIES
+  URL "https://github.com/FFmpeg/nv-codec-headers/"
+  DESCRIPTION "FFmpeg version of headers required to interface with Nvidias codec APIs.")
```
Comment (RytoEX) [inline cmake/finders/FindFFnvcodec.cmake:96]:
```suggestion
  DESCRIPTION "FFmpeg version of headers required to interface with NVIDIA's codec APIs.")
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: cmake/Modules/FindFFnvcodec.cmake (hunk 5 above)
Comment (tytan652) [reply cmake/Modules/FindFFnvcodec.cmake]: Change done.

**Pull request merged**